### DISCUSSION TITLE
Bitbucket server / generic SSH

### DISCUSSION THREAD

**Discussion by UbiquitousBear** [terrakube-io/terrakube#579] [Ideas]
I'm using Bitbucket server (or data centre as the products are similar) and would like to use Terrakube. Currently Terrakube doesn't support non cloud based VCS providers. 

Based on my understanding of the VCS code in the `api`, access is token-based, and I can envisage PATs (Personal Access Tokens) could suffice here?

What are your thoughts?

**Comment by alfespa17** [comment]
Hello @UbiquitousBear 

I would like to add support for non cloud VCS provider but I don't have a way to test it I guess it will be the same logic that we are currently using but adding some filed for the hostmane.

For now the alternative will be to use SSH to connect to your server.

You could check this

https://docs.terrakube.io/user-guide/vcs-providers/ssh

You should be able to use modules and workspaces using that option

**Reply by UbiquitousBear** [reply]
Perfect, that works!
